plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -688,20 +688,18 @@ public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle ha
                 .map(this::quoted)
                 .collect(joining(", "));
         String insertSql = format("INSERT INTO %s (%s) SELECT %s FROM %s", targetTable, columnNames, columnNames, temporaryTable);
-        String cleanupSql = "DROP TABLE " + temporaryTable;
 
         try (Connection connection = getConnection(session, handle)) {
             execute(connection, insertSql);
         }
         catch (SQLException e) {
             throw new TrinoException(JDBC_ERROR, e);
         }
-
-        try (Connection connection = getConnection(session, handle)) {
-            execute(connection, cleanupSql);
-        }
-        catch (SQLException e) {
-            log.warn(e, "Failed to cleanup temporary table: %s", temporaryTable);
+        finally {
+            JdbcTableHandle temporaryTableHandle = new JdbcTableHandle(
+                    new SchemaTableName(handle.getSchemaName(), handle.getTemporaryTableName()),
```

Comment:
if `handle.getTemporaryTableName()` represents a remote name then to use it as Trino name you need to use `io.trino.plugin.jdbc.mapping.IdentifierMapping#fromRemoteTableName`

That is the motivation behind `RemoteTableName`. Now remote name and Trino names are plain `Strings` so it is easy to confuse them.